tests/utils/utils.go

```diff
@@ -128,7 +275,43 @@ func CreateSendCrossChainMessageTransaction(
 		Data:      data,
 	})
 
-	return signTransaction(tx, fundedKey, source.ChainIDInt)
+	return SignTransaction(tx, fundedKey, source.ChainIDInt)
+}
+
+func CreateRetryMessageExecutionTransaction(
+	ctx context.Context,
+	subnetInfo SubnetTestInfo,
+	originChainID ids.ID,
+	message teleportermessenger.TeleporterMessage,
+	fundedAddress common.Address,
+	fundedKey *ecdsa.PrivateKey,
+	teleporterContractAddress common.Address,
+) *types.Transaction {
+	teleporterABI, err := teleportermessenger.TeleporterMessengerMetaData.GetAbi()
+	Expect(err).Should(BeNil())
+
+	data, err := teleporterABI.Pack("retryMessageExecution", originChainID, message)
+	Expect(err).Should(BeNil())
+
+	// TODO: replace with actual number of signers
+	gasLimit, err := gasUtils.CalculateReceiveMessageGasLimit(10, message.RequiredGasLimit)
+	Expect(err).Should(BeNil())
+
+	gasFeeCap, gasTipCap, nonce := CalculateTxParams(ctx, subnetInfo, fundedAddress)
+
+	// Send a transaction to the Teleporter contract
+	tx := types.NewTx(&types.DynamicFeeTx{
+		ChainID:   subnetInfo.ChainIDInt,
+		Nonce:     nonce,
+		To:        &teleporterContractAddress,
+		Gas:       gasLimit,
+		GasFeeCap: gasFeeCap,
+		GasTipCap: gasTipCap,
+		Value:     DefaultTeleporterTransactionValue,
+		Data:      data,
+	})
+
+	return SignTransaction(tx, fundedKey, subnetInfo.ChainIDInt)
```

Comment:
This should be refactored to use t`eleportermessenger.NewTeleporterMessenger` from the abi-bindings

Reply:
Added `PackRetryMessageExecution` to package `teleportermessenger`